### PR TITLE
fix: clean up test project on signal, fixes #6111

### LINKED ISSUE #6111: `ddev debug test` may have conflicts when running sequentially (and `ddev debug testcleanup` have no effect)
### Is there an existing issue for this?

- [X] I have searched the existing issues

From Discord:
https://discord.com/channels/664580571770388500/1230114214182522922/1230114214182522922

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```
[COPY-PASTE HERE THE OUTPUT OF `ddev debug test`]
```
</details>

### Expected Behavior

Each `ddev debug test` should succeed and cleanup after itself.

### Actual Behavior

1. `host port 60001 has already been allocated`:
```
...
Configuring a 'php' project with docroot 'web' at /home/stas/code/ddev/tryddevproject-5008/web
could not write DDEV config file /home/stas/code/ddev/tryddevproject-5008/.ddev/config.yaml: host port 60001 has already been allocated to project tryddevproject-27049
...
```

2. There is `ddev stop --unlist` at the end of the script, which does not clean up Docker resources.

### Steps To Reproduce

Run `ddev debug test`
Abort the test before `ddev stop --unlist`:
```
======== Thanks for running the diagnostic! ========
Running ddev launch in 3 seconds
Running ddev launch
Waiting 10 seconds to run ddev stop --unlist
^C
```
Run `ddev debug test` again, see an error

### Anything else?

Maybe it is better to add `ddev debug testcleanup` at script start and script end, because when you run `ddev stop --unlist`, the test project cannot be removed with cleanup command.

### PR DESCRIPTION
**Pull request by rfay**
## The Issue

* #6111 
* Adding build-time connectivity test

## How This PR Solves The Issue

* Catch signal and clean up on exit
* Check docker build-time connectivity to internet

## Manual Testing Instructions

- [ ] Ctrl-C during run, it should clean up or give a good try
- [ ] It should completely clean up after a normal run
- [ ] If not internet is available, or google.com not accessible, the failure should show

### CODE REVIEW

File: cmd/ddev/cmd/scripts/test_ddev.sh [hunk 1]
```diff
@@ -65,11 +65,22 @@ header "Creating dummy project named ${PROJECT_NAME} in ${PROJECT_DIR}"
...
-ddev config --project-type=php --docroot=web --disable-upload-dirs-warning --host-db-port=60001 --host-https-port=60002 --host-webserver-port=60003 || (printf "\n\nPlease run 'ddev debug test' in the root of the existing project where you're having trouble.\n\n" && exit 4)
+
+function cleanup {
+  printf "\n\nCleanup: deleting test project ${PROJECT_NAME}\n"
+  ddev delete -Oy ${PROJECT_NAME}
+  printf "\nPlease remove the files from this test with 'rm -r ${PROJECT_DIR}'\n"
+}
+
+ddev config --project-type=php --docroot=web --disable-upload-dirs-warning || (printf "\n\nPlease run 'ddev debug test' in the root of the existing project where you're having trouble.\n\n" && exit 4)
 printf "\nhost_mailpit_port: 60004\n" >.ddev/config.local.yaml
```
Comment (stasadev) [inline cmd/ddev/cmd/scripts/test_ddev.sh:76]: `60004` should also be removed?

**Review by stasadev — CHANGES_REQUESTED** [review]
Tested
- [x] Ctrl-C during run, it should clean up or give a good try
- [x] It should completely clean up after a normal run
- [x] If not internet is available, or google.com not accessible, the failure should show

Thanks!

---

Minor notes:

1. There is one more hard-coded port `60004`.

2. `$router_http_port` and `$router_https_port` are not used in the script:

https://github.com/ddev/ddev/blob/d358586edb9d6eed0625115887900c0f8d5f3020/cmd/ddev/cmd/scripts/test_ddev.sh#L83-L84

3. A typo in `healtcheck` > `healthcheck`
https://github.com/ddev/ddev/blob/d358586edb9d6eed0625115887900c0f8d5f3020/cmd/ddev/cmd/scripts/test_ddev.sh#L164

**Comment by rfay** [comment]
Thanks for the careful review!
* Removed extraneous environment variables
* Added curl of the webserver http bind port

File: cmd/ddev/cmd/scripts/test_ddev.sh [hunk 2]
```diff
@@ -105,7 +103,7 @@ header "DDEV global info"
 ddev config global | (grep -v "^web-environment" || true)
 
 header "DOCKER provider info"
-printf "docker client location: $(which docker)\n\n"
+printf "docker client location: ls -l \"$(which docker)\"\n\n"
```
Comment (stasadev) [inline cmd/ddev/cmd/scripts/test_ddev.sh:106]: Not sure why `ls -l` is added here as text and not as a command.

File: cmd/ddev/cmd/scripts/test_ddev.sh [hunk 3]
```diff
@@ -189,19 +187,23 @@ DDEV_DEBUG=true ddev start -y || ( \
...
+host_http_url=$(ddev describe -j | docker run -i --rm ddev/ddev-utilities jq -r  '.raw.services.web.host_http_url' 2>/dev/null)
 http_url=$(ddev describe -j | docker run -i --rm ddev/ddev-utilities jq -r  '.raw.httpURLs[0]' 2>/dev/null)
 https_url=$(ddev describe -j | docker run -i --rm ddev/ddev-utilities jq -r  '.raw.httpsURLs[0]' 2>/dev/null)
 
 header "Curl of site from inside container"
 ddev exec curl --fail -I http://127.0.0.1
 
-header "curl -I of ${http_url} from outside"
+header "curl -I of ${host_http_url} (web container http docker bind port) from outside"
+curl --fail -I ${host_http_url}
```
Comment (stasadev) [inline cmd/ddev/cmd/scripts/test_ddev.sh:198]: Do we also need to curl for `${host_https_url}` (there is not such variable yet)?

File: cmd/ddev/cmd/scripts/test_ddev.sh [hunk 4]
```diff
@@ -189,19 +187,23 @@ DDEV_DEBUG=true ddev start -y || ( \
...
+
+header "curl -I of ${http_url} (router http URL) from outside"
 curl --fail -I "${http_url}"
 
-header "Full curl of ${http_url} from outside"
+header "Full curl of ${http_url} (router https URL) from outside"
```
Comment (stasadev) [inline cmd/ddev/cmd/scripts/test_ddev.sh:203]: Please review these curls, because it says `router https URL` but runs it for `${http_url}`

**Review by stasadev — CHANGES_REQUESTED** [review]

File: cmd/ddev/cmd/scripts/test_ddev.sh (hunk 3 above)
Comment (rfay) [reply cmd/ddev/cmd/scripts/test_ddev.sh:198]: The host_https_url is pretty useless, doesn't really demonstrate anything at all, and we don't really want people to use it. I don't think.

File: cmd/ddev/cmd/scripts/test_ddev.sh (hunk 4 above)
Comment (rfay) [reply cmd/ddev/cmd/scripts/test_ddev.sh]: Thanks, fixed

**Comment by rfay** [comment]
Thanks for your eagle eye!

**Review by stasadev — APPROVED** [review]
Tested again, all good 👍

**Pull request merged**